python/ml4ir/applications/ranking/tests/test_cyclic_lr_schedule.py

```diff
@@ -0,0 +1,58 @@
+import unittest
+import warnings
+from ml4ir.base.features.preprocessing import *
+from ml4ir.base.model.optimizers import cyclic_learning_rate
+
+warnings.filterwarnings("ignore")
+
+INPUT_DIR = "/ml4ir/applications/ranking/tests/data/"
+KEEP_ADDITIONAL_INFO = 0
+NON_ZERO_FEATURES_ONLY = 0
+
+
+class TestRanklibConversion(unittest.TestCase):
+    def test_cyclic_tri_lr(self):
+        """Create a triangular cyclic learning rate"""
+        gold = [0.1,0.28000003,0.45999995,0.64000005,0.81999993,1.0,0.81999993,0.64000005,0.45999995,0.28000003,
+                0.1,0.28000003,0.46000007,0.6399999,0.81999993,1.0,0.81999993,0.6399999,0.46000007,0.28000003,
+                0.1,0.27999982,0.46000007,0.6399999,0.8200002]
+        lrs = cyclic_learning_rate.TriangularCyclicalLearningRate(
+            initial_learning_rate=0.1,
+            maximal_learning_rate=1.0,
+            step_size=5,
+        )
+        for i in range(25):
+            assert round(float(lrs(i).numpy()), 5) == round(gold[i], 5)
```

Comment:
@mohazahran This is fine for now, but you can also look at [numpy.isclose()](https://numpy.org/doc/stable/reference/generated/numpy.isclose.html). It has absolute and relative thresholds you can set.